src/mergeDiff.js

```diff
@@ -40,7 +41,9 @@ function mergeDiffArr(arrA, arrB, collB, indexA, indexB, onRemove, accum) {
   return mergeDiffArr(arrA, arrB, collB, indexA + 1, indexB, onRemove, accum);
 }
 
-export default function mergeDiff(a, b, onRemove) {
+export default function mergeDiff(a: Object,
+                                  b: Object,
+                                  onRemove: Function): Object {
```

Comment:
We can specify `onRemove`’s parameter types and return type. It looks like `mergeDiff` would work on any two objects: would you like to define it only for certain types of objects? It’s possible to define a Style type, for example, and then say that functions only work on Style.